### DISCUSSION TITLE
spawn actors while creating machine

### DISCUSSION THREAD

**Discussion by anonymouscatcher** [statelyai/xstate#3683] [Q&A] [answered]
Is there any way to spawn actor in context without calling assign? currently I'm facing some issue like `actor is not object`, the reason is actor is not spawned yet and it's null even when I spawn it in `entry`!

```js
createMachine({
 // do this
  context: {
     actor1: spawn(xxx)
   }
  // instead of this
  entry: [
    assign({
      actor1() {
        return spawn(xxx)
      }
    })
  ]
})
```

**Comment by Andarist** [comment, marked as answer]
As long as your `xxx` actor is a machine this should work in v4:
```ts
createMachine({
  context: () => ({
     actor1: spawn(xxx)
   })
})
```

**Comment by anonymouscatcher** [comment]
🤦‍♂️ I couldn't find this in documents! thanks
